File: tests/semanticdb/expect/MetacJava.scala
package example

import com.javacp

class MetacJava {
  javacp.MetacJava.StaticInner.isStatic()
  new javacp.MetacJava.StaticInner().isNotStatic()
  val inner = new javacp.MetacJava()
  val overload1 = new inner.Overload1()
  val overload2 = new inner.Overload2()
  inner.overload(new overload1.A())
  inner.overload(new overload2.A())
  val staticInner = new javacp.MetacJava.StaticInner()
  val nonStatic = new staticInner.NonStatic()
  nonStatic.method(nonStatic)
  javacp.MetacJava.overload(new javacp.MetacJava.Overload3.A())
  val interface: javacp.Interface = null
  val coin: javacp.Coin = javacp.Coin.PENNY
  val entry: java.util.Map.Entry[Int, Int] = null
}

File: tests/semanticdb/javacp/com/javacp/MetacJava.java
package com.javacp;

public class MetacJava {
    public static class StaticInner {
        public void isNotStatic() {}
        public static void isStatic() {}
        public class NonStatic {
            public void method(NonStatic e) {}
        }
    }
    public class Overload1 { public class A {} }
    public static class Overload3 {
        public static class A {}
    }
    public class Overload2 { public class A {} }
    public void overload(Overload1.A a) {}
    // NOTE: Overload3 is intentionally placed before Overload2 in order to test
    // that methods are sorted by whether they're static or not.
    public static void overload(Overload3.A a) {}
    public void overload(Overload2.A a) {}
}
